app/controllers/links_controller.rb

```diff
@@ -150,55 +150,58 @@ def create
   def show
     return redirect_to custom_domain_coffee_path if @product.native_type == Link::NATIVE_TYPE_COFFEE
     ActiveRecord::Base.connection.stick_to_primary!
-    # Force a preload of all association data used in rendering
```

Comment:
I am merely moving this code into the `format.html` block since it is only used in the related HTML template, thus making the other request formats (JSON. any) more performant since it is no longer executed for those.